modules/widgets/milestone/milestone.php

```diff
@@ -183,59 +217,240 @@ function widget( $args, $instance ) {
 			$diff = (int) floor( $milestone - $now );
 		}
 
-		echo $args['before_widget'];
-
-		$title = apply_filters( 'widget_title', $instance['title'] );
-		if ( ! empty( $title ) ) {
-			echo $args['before_title'] . $title . $args['after_title'];
-		}
-
-		echo '<div class="milestone-content">';
+		$data['diff'] = $diff;
+		$data['unit'] = $this->get_unit( $diff, $instance['unit'] );
+
+		// Setting the refresh counter to equal the number of seconds it takes to flip a unit
+		$refresh_intervals = array(
+			0, // should be YEAR_IN_SECONDS, but doing setTimeout for a year doesn't seem to be logical
+			0, // same goes for MONTH_IN_SECONDS,
+			DAY_IN_SECONDS,
+			HOUR_IN_SECONDS,
+			MINUTE_IN_SECONDS,
+			1
+		);
 
-		echo '<div class="milestone-header">';
-		echo '<strong class="event">' . esc_html( $instance['event'] ) . '</strong>';
-		echo '<span class="date">' . esc_html( date_i18n( __( 'F jS, Y', 'jetpack' ), $milestone ) ) . '</span>';
-		echo '</div>';
+		$data['refresh'] = $refresh_intervals[ array_search( $data['unit'], $this->available_units ) ];
+		$data['milestone'] = $milestone;
 
 		if ( ( 1 > $diff ) && ( 'until' === $type ) ) {
-			echo '<div class="milestone-message">' . $instance['message'] . '</div>';
+			$data['message'] = $instance['message'];
+			$data['refresh'] = 0; // No need to refresh, the milestone has been reached
 		} else {
+			$interval_text = $this->get_interval_in_units( $diff, $data['unit'] );
+			$interval = intval( $interval_text );
+
 			if ( 'since' === $type ) {
-				$text = __( 'ago.', 'jetpack' );
+
+				switch ( $data['unit'] ) {
+					case 'years':
+						$data['message'] = sprintf(
+							_n(
+								'<span class="difference">%1$s</span> <span class="label">year ago.</span>',
+								'<span class="difference">%1$s</span> <span class="label">years ago.</span>',
+								$interval,
+								'jetpack'
+							),
+							$interval_text
+						);
+					break;
+					case 'months':
+						$data['message'] = sprintf(
+							_n(
+								'<span class="difference">%1$s</span> <span class="label">month ago.</span>',
```

Comment:
For better readability/ease of translation we can change all `%1$s` to just `%s` when there is just 1 variable in the string. As far as I can tell this is the case for all strings in this function.